src/transaction/client.rs

```diff
@@ -15,7 +15,23 @@ use tikv_client_proto::{kvrpcpb, pdpb::Timestamp};
 
 const SCAN_LOCK_BATCH_SIZE: u32 = 1024; // TODO: cargo-culted value
 
-/// The TiKV transactional `Client` is used to issue requests to the TiKV server and PD cluster.
+/// The TiKV transactional `Client` is used to interact with TiKV using transactional (MVCC) requests.
+///
+/// A [`Transaction`](crate::transaction::Transaction) provides a SQL-like interface.
+/// It begins with a [`begin`](Client::begin) or [`begin_pessimistic`](Client::begin_pessimistic) request
+/// and ends with a `rollback` or `commit` request.
+/// If a `Transaction` is dropped before it's rolled back or committed, it is automatically rolled back.
+///
+/// Transaction supports optimistic and pessimistic modes, for mroe deatils, check our
+/// [SIG-transaction](https://github.com/tikv/sig-transaction/tree/master/doc/tikv#optimistic-and-pessimistic-transactions).
+///
+/// Besides transaction, the client provides some utility methods:
+/// - `gc`: execute a GC process which clear stale data. It is not stablized yet.
+/// - `current_timestamp`: get the current `Timestamp`.
+/// - `snapshot`: get the [`Snapshot`](crate::transaction::Snapshot) of the database at a certain timestamp.
```

Comment:
Are these public?

Reply:
They are

Reply:
Do they need to be?

Reply:
For now, I think they are intended to be public because they are not used by other parts.
`snapshot` is a wrapper of transaction to make it read-only. If it's not public we can remove it.
`gc` needs to be manually triggered by users and requires a `Timestamp`.
`current_timestamp` is the easiest way for users to get a timestamp. We don't provide convenience methods for users to calculate Timestamps. Maybe we should add these in the future.

Reply:
Do users ever need to supply timestamps? I thought they were internal to the transaction? I guess we need to consider the lower level API.

Does Snapshot offer anything that Transaction doesn't?

Reply:
Timestamps are needed for GC and snapshot. 

In GC users should supply a safe point. In a TiDB cluster there will be only one node that performs GC and it controls when to do it. In a TiKV client we cannot guarantee this, so I think it should be the user's choice.

`Snapshot` lets you read from any given timestamp, while `Transaction` cannot. Also the timestamp is related to GC safe point, which could be another supportive reason for users providing safe point for GC.

Reply:
OK, thanks!